core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -514,12 +520,13 @@ sealed abstract class Doc extends Product with Serializable {
     @tailrec
     def loop(pos: Int, lst: List[(Int, Doc)], max: Int): Int = lst match {
       case Nil => math.max(max, pos)
-      case (i, Empty) :: z => loop(pos, z, max)
+      case (i, Empty | Fail) :: z => loop(pos, z, max)
```

Comment:
seems like we should not just silently skip `Fail` to me. Can you explain in a comment why we should?

Seems to me that we want `Union(Fail, a) == a` and otherwise we don't expect to see `Fail`. Actually, I would say `Fail` should actually throw in rendering since by invariant we expect it to never come up.